src/components/tx-flow/SafeTxProvider.tsx

```diff
@@ -45,13 +45,12 @@ const SafeTxProvider = ({ children }: { children: ReactNode }): ReactElement =>
   // Signed txs cannot be updated
   const isSigned = safeTx && safeTx.signatures.size > 0
 
-  // Recommended nonce and safeTxGas
+  // Recommended nonce
   const recommendedNonce = useRecommendedNonce()
-  const recommendedSafeTxGas = useSafeTxGas(safeTx)
 
   // Priority to external nonce, then to the recommended one
   const finalNonce = isSigned ? safeTx?.data.nonce : nonce ?? recommendedNonce ?? safeTx?.data.nonce
-  const finalSafeTxGas = isSigned ? safeTx?.data.safeTxGas : safeTxGas ?? recommendedSafeTxGas ?? safeTx?.data.safeTxGas
+  const finalSafeTxGas = isSigned ? safeTx?.data.safeTxGas : safeTxGas ?? safeTx?.data.safeTxGas
```

Comment:
Am I the only one who thinks that this is terrible to read?

Isn't:
```

let finalNonce;

if (isSigned) {
  finalNonce = safeTx?.data.nonce;
} else if (nonce) {
  finalNonce = nonce;
} else if (recommendedNonce) {
  finalNonce = recommendedNonce;
} else {
  finalNonce = safeTx?.data.nonce;
}
```

or 

```
let finalNonce;
if (isSigned) {
  finalNonce = safeTx?.data.nonce;
} else {
  finalNonce = nonce || recommendedNonce || safeTx?.data.nonce;
}
```

I know it is on multiple lines, but I can easily see what's going on. Where with ? : ?? ?? ?? I have to be lucky to see where it ends :D

Reply:
True, I think that ternary ran out of control :)